packages/react/src/primitives/Button/Button.tsx

```diff
@@ -1,37 +1,47 @@
 import classNames from 'classnames';
+import * as React from 'react';
+
 import { ComponentClassNames } from '../shared/constants';
-import { ButtonProps, Primitive } from '../types';
+import { ButtonProps, PrimitiveWithForwardRef } from '../types';
 import { View } from '../View';
 
-export const Button: Primitive<ButtonProps, 'button'> = ({
-  className,
-  children,
-  isFullWidth = false,
-  isDisabled,
-  isLoading,
-  loadingText = '',
-  size,
-  type = 'button',
-  variation,
-  ...rest
-}) => (
-  <View
-    as="button"
-    className={classNames(
-      ComponentClassNames.Button,
-      ComponentClassNames.FieldGroupControl,
-      className
-    )}
-    data-fullwidth={isFullWidth}
-    data-loading={isLoading}
-    data-size={size}
-    data-variation={variation}
-    isDisabled={isDisabled || isLoading}
-    type={type}
-    {...rest}
-  >
-    {isLoading && loadingText ? <span>{loadingText}</span> : children}
-  </View>
-);
+export const ButtonInner: PrimitiveWithForwardRef<ButtonProps, 'button'> = (
```

Comment:
We may want to rename `PrimitiveWithForwardRef` to just `Primitive` when we've migrated all the other components. At this point I didn't want to touch the other primitives.

Reply:
cool!